concordium-consensus/src/Concordium/Kontrol/Bakers.hs

```diff
@@ -190,26 +190,26 @@ computeBakerStakesAndCapital poolParams activeBakers passiveDelegators = BakerSt
     capLimit = takeFraction (theCapitalBound capitalBound) totalCapital
     makeBakerStake ActiveBakerInfo{..} poolCap =
         ( activeBakerInfoRef,
-          minimum
-            [ poolCap,
-              applyLeverageFactor leverage activeBakerEquityCapital,
-              capLimit
-            ]
+          if activeBakerIsSuspended
+            then 0
+            else
+                minimum
+                    [ poolCap,
+                      applyLeverageFactor leverage activeBakerEquityCapital,
+                      capLimit
+                    ]
```

Comment:
I changed this as described. I don't have a hard opinion on this, I think both approaches suffer a bit from being not very explicit.
> I'm not 100% certain that we never rely on validators' stakes being non-zero.

Are we 100% sure that we never rely on validators being present in `bakerStakes`?

> Even 0-weighted validators could end up as finalizers, and potentially end up signing blocks even though their signatures contribute nothing. This seems weird to me.

Agreed.

> If we remove the suspended validators from the epoch bakers/capital distribution, that means that when they become un-suspended then their missed block counters will be reset to 0, which I think is what we want. Maybe?

Yes, this is what we want. I'm not clear however why this would happen as a consequence from them being removed from the epoch bakers/capital distribution.

Reply:
:+1:  makes sense, thanks!